src/main/resources/META-INF/rewrite/jakarta-faces-4.yml

```diff
@@ -448,3 +449,77 @@ recipeList:
       oldFullyQualifiedTypeName: jakarta.faces.bean.ViewScoped
       newFullyQualifiedTypeName: jakarta.faces.view.ViewScoped
       ignoreDefinition: true
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.jakarta.UpgradeFacesOpenSourceLibraries
+displayName: Upgrade Faces open source libraries
+description: Upgrade PrimeFaces, OmniFaces, and MyFaces libraries to Jakarta EE10 versions.
+recipeList:
+  - org.openrewrite.maven.ChangeDependencyClassifier:
+      groupId: org.primefaces
+      artifactId: primefaces
+      newClassifier: jakarta
+  - org.openrewrite.maven.ChangeDependencyGroupIdAndArtifactId:
+      oldGroupId: org.primefaces
+      oldArtifactId: primefaces
+      newGroupId: org.primefaces
+      newArtifactId: primefaces
+      newVersion: 13.0.x
+  - org.openrewrite.maven.ChangeManagedDependencyGroupIdAndArtifactId:
+      oldGroupId: org.primefaces
+      oldArtifactId: primefaces
+      newGroupId: org.primefaces
+      newArtifactId: primefaces
```

Comment:
There's no need to repeat the groupId or artifactId when the value remains the same. Just the old values is then enough. Sorry I didn't spot that the first time around!

Reply:
Also: you're using the Maven variant of the recipe, whereas in. Rewrite-java-dependencies we also have one to change both Maven and gradle dependency versions

Reply:
https://docs.openrewrite.org/recipes/java/dependencies/upgradedependencyversion

Reply:
Ok let me look at that one you mentioned

Reply:
OK I updated to the UpgradeDependencyVersion but for ChangedManagedVersion it does NOT work without the `new` values it says they are required and errors.

Also I am updating the Classifier to `jakarta` but in main rewrite there is no way to change a "Managed Dependency" classifier it looks like that Recipe is missing?

Reply:
I opened this ticket: https://github.com/openrewrite/rewrite/issues/3830